canvas_sdk/data/base.py

```diff
@@ -0,0 +1,15 @@
+import json
+
+from pydantic import BaseModel, ConfigDict
+
+from canvas_sdk.utils.json import JSONEncoder
+
+
+class PluginModel(BaseModel):
+    model_config = ConfigDict(strict=True, validate_assignment=True)
```

Comment:
The more I think about it the more I think we shouldn't do any validation except attribute type validation or when returning effects. That is, don't let people assign a string to an int-typed attribute, and don't care about the completeness of an object unless we're trying to return an effect that requires some amount of enforced completeness.

Reply:
I just pushed up changes that will validate before returning an effect.